independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -455,6 +458,22 @@ private void addClassLevelBindings(ClassInfo classInfo, Collection<AnnotationIns
         }
     }
 
+    private void addConstructorLevelBindings(ClassInfo classInfo, Collection<AnnotationInstance> bindings) {
+        MethodInfo constructor;
+        Optional<Injection> constructorWithInject = getConstructorInjection();
+        if (constructorWithInject.isPresent()) {
+            constructor = constructorWithInject.get().target.asMethod();
+        } else {
+            constructor = classInfo.method(Methods.INIT);
+        }
+        if (constructor != null) {
+            beanDeployment.getAnnotations(constructor).stream()
```

Comment:
Well, I'm not a big fan of using streams when a simple for loop is good enough:
```
for (AnnotationInstance a : beanDeployment.getAnnotations(constructor)) {
     if (beanDeployment.getInterceptorBinding(a.name()) != null && bindings.stream().noneMatch(e -> e.name().equals(a.name()))) {
        bindings.add(a);
     }
}
```
But it's probably because I'm too old ;-). Anyway, at least `.forEach(a -> bindings.add(a))` could be replaced with a method reference `.forEach(bindings::add)`.